www/src/components/account/billing/PaymentForm.tsx

```diff
@@ -567,36 +563,50 @@ function AddressForm({
   )
 }
 
-function SelectPaymentMethod() {
+function SelectUpgradePaymentMethod() {
   const { setFormState, plan } = usePaymentForm()
-  const { defaultPaymentMethod, paymentMethods } = useBillingSubscription()
-  const [upgradeSuccess, setUpgradeSuccess] = useState(false)
+  const { defaultPaymentMethod, paymentMethods, refetch }
+    = useBillingSubscription()
+  const [error, setError] = useState<Error | StripeError | undefined>()
   const { proPlatformPlan, proYearlyPlatformPlan }
     = useContext(PlatformPlansContext)
   const navigate = useNavigate()
   const planId
     = plan === 'yearly' ? proYearlyPlatformPlan.id : proPlatformPlan.id
 
   // Upgrade mutation
-  const [upgradeMutation, { loading, error }] = useUpgradeSubscription(
-    { planId },
-    (result, nextPath) => navigate(`${nextPath}&payment_intent_client_secret=${result.paymentIntent.client_secret}`),
-    () => null,
-    () => setUpgradeSuccess(true)
-  )
+  const [upgradeMutation, { loading }] = useCreatePlatformSubscriptionMutation({
```

Comment:
why not use the `useUpgradeSubscription` function here?

Reply:
This way I can just redirect to `<ConfirmPayment>` to handle both `stripe.confirmPayment` and all the error/success messaging.

Reply:
Oh i think i see the difference here.